sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -214,50 +220,54 @@ private Response<?> receiveResponse(HttpRequest httpRequest, HttpURLConnection c
+            serverSentResult = processTextEventStream(connection.getInputStream(), listener);
 
-            if (options != null) {
-                responseBodyMode = options.getResponseBodyMode();
+            if (Thread.currentThread().isInterrupted() || !retryReconnect(serverSentResult, httpRequest)) {
+                listener.onError(serverSentResult.getException());
+            } else {
+                this.send(httpRequest).close();
             }
+        }
 
-            if (responseBodyMode == null) {
-                HttpHeader contentType = httpResponse.getHeaders().get(CONTENT_TYPE);
-
-                if (httpRequest.getHttpMethod() == HEAD) {
-                    responseBodyMode = IGNORE;
-                } else if (contentType != null && APPLICATION_OCTET_STREAM
-                    .regionMatches(true, 0, contentType.getValue(), 0, APPLICATION_OCTET_STREAM.length())) {
+        ResponseBodyMode responseBodyMode = null;
 
-                    responseBodyMode = STREAM;
-                } else {
-                    responseBodyMode = BUFFER;
-                }
-            }
+        if (options != null) {
+            responseBodyMode = options.getResponseBodyMode();
+        }
 
-            switch (responseBodyMode) {
-                case IGNORE:
-                    HttpResponseAccessHelper.setBody(httpResponse, BinaryData.EMPTY);
+        if (responseBodyMode == null) {
+            responseBodyMode = determineResponseBodyMode(httpRequest, httpResponse.getHeaders());
+        }
 
-                    // Close the response InputStream rather than using disconnect. Disconnect will close the Socket
-                    // connection when the InputStream is still open, which can result in keep alive handling not being
-                    // applied.
-                    connection.getInputStream().close();
+        switch (responseBodyMode) {
+            case IGNORE:
+                HttpResponseAccessHelper.setBody(httpResponse, BinaryData.EMPTY);
 
-                    break;
-                case STREAM:
-                    streamResponseBody(httpResponse, connection);
+                // Close the response InputStream rather than using disconnect. Disconnect will close the Socket
+                // connection when the InputStream is still open, which can result in keep alive handling not being
+                // applied.
+                connection.getInputStream().close();
 
-                    break;
-                case BUFFER:
-                case DESERIALIZE: // Deserialization will occur at a later point in HttpResponseBodyDecoder.
-                default:
+                break;
+            case STREAM:
+                streamResponseBody(httpResponse, connection);
+                break;
+            case BUFFER:
+            case DESERIALIZE:
+                // Deserialization will occur at a later point in HttpResponseBodyDecoder.
+                if (isTextEventStream(responseHeaders)) {
```

Comment:
Does it make sense for SSE to reach the handler for the response type as it will either be a `stream` response that's already consumed, `buffer` or `deserialize` that is the content of the last received event, or an empty eager read in the default case.

Reply:
Updated the `stream` scenario for SSE data to match `buffer` and `deserialize`.
Also, the `serverSentResult.getData()` returns the accumulated data, [here](https://github.com/Azure/azure-sdk-for-java/pull/39835/files#diff-a1971465bccbdd9c59a6dd22f726e9c1393909b6f3a6a6583c68719c39f314f4R26)